src/core/net/ip6.cpp

```diff
@@ -1313,7 +1316,13 @@ bool Ip6::ShouldForwardToThread(const MessageInfo &aMessageInfo) const
     else if (IsOnLink(aMessageInfo.GetSockAddr()))
     {
         // on-link global address
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+        ExitNow(rval = (aFromNcpHost ||
+                        !Get<BackboneRouter::Manager>().ShouldForwardDuaToBackbone(aMessageInfo.GetSockAddr())));
+#else
+        OT_UNUSED_VARIABLE(aFromNcpHost);
```

Comment:
nit: move to the beginning.

https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md

> Unused parameters
All unused parameters shall be declared as such using the OT_UNUSED_VARIABLE macro at the top of a function or method before all local variable declarations.